### PR TITLE
feat(chart): Added annotation: to api.service and default api.service.annotations value to values.yaml

### PR DESCRIPTION
**Pull request by ed-boykin**
Added a section to charts/kargo/templates/api/service.yaml to add 0 to many annotations to the api service.
When hosting Kargo in a managed K8s system, in this case AKS, an annotation can be added to create an internal or externally facing managed azure load balancer for ingress.

### CODE REVIEW

**Comment by krancour** [comment]
Hi @ed-boykin. Thanks for this!

I understand your need, but want to suggest a more general-purpose approach to this so that arbitrary annotations can be added to any/all Kargo pods, as there may be reasons beyond those you are encountering to do such a thing.

If I may suggest, @MPritsch has taken a very nice approach to doing something similar for adding arbitrary env vars to any/all Kargo pods over in #1874.

If you could emulate that approach, I think this would be something we could get into v0.6.0.

Thanks again!

File: charts/kargo/values.yaml [hunk 1]
```diff
@@ -123,6 +123,8 @@ api:
     type: ClusterIP
     ## @param api.service.nodePort [nullable] Host port the `Service` will be mapped to when `type` is either `NodePort` or `LoadBalancer`. If not specified, Kubernetes chooses.
     # nodePort:
+    ## @param api.service.annotations Annotations to add to the api service.
```
Comment (krancour) [inline charts/kargo/values.yaml:126]:
```suggestion
    ## @param api.service.annotations Annotations to add to the API server's service. 
```

Comment (krancour) [reply charts/kargo/values.yaml:126]: With the change above, this LGTM. You do need to [run codegen](https://kargo.akuity.io/contributor-guide/hacking-on-kargo#executing-code-generation) after any edits to `values.yaml`.

If you prefer, I can run it for you.

Thanks!

Comment (ed-boykin) [reply charts/kargo/values.yaml:126]: Thanks Kent. If you wouldn't mind running it. Go isn't a strength for me and I'd probably not do it right.

**Review by krancour — APPROVED** [review]

**Comment by krancour** [comment]
Thank you @ed-boykin!

**Pull request merged**